[new] Added RiTa@1.1.40 w/ git auto update

CyrusSUEN opened the pull request:
Referring to ticket https://github.com/cdnjs/cdnjs/issues/8442. Thank you so much for your hard work here at cdnjs!

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/dhowe/RiTaJS 
 * Official website (optional, not the repository): http://rednoise.org/rita/
 * NPM package url (optional): https://www.npmjs.com/package/rita
 * GitHub / Bitbucket popularity (required): 
   - Count of stars: 35
   - Count of watchers: 6
   - Count of forks: 10
 * NPM download stats (optional):
   - Downloads in the last day: 1
   - Downloads in the last week: 123
   - Downloads in the last month: 662

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

PeterDaveHello commented:
@cdnjs/intern2 please review this PR.

pvnr0082t commented on ajax/libs/rita/package.json:
```diff
@@ -0,0 +1,33 @@
+  "tonicExampleFilename": "tonic.js",
+  "optionalDependencies": {
+    "yamljs": "^0.2.3"
+  },
```
@CyrusSUEN Let's remove `tonicExampleFilename`, `optionalDependencies` fields.

pvnr0082t commented on ajax/libs/rita/package.json:
```diff
@@ -0,0 +1,33 @@
+  "version": "1.1.40",
+  "author": "Daniel C. Howe <daniel@rednoise.org>",
+  "description": "RiTa: A toolkit for natural language and computational literature",
+  "homepage": "http://rednoise.org/rita",
```
I think we can use `https` instead of `http` here.

CyrusSUEN replied:
Thank you sir! This is a nice catch!

pvnr0082t commented on ajax/libs/rita/package.json:
```diff
@@ -0,0 +1,33 @@
+      ]
+    }
+  ],
+  "filename": "/rita-full.min.js"
```
I think the main file should be `"rita.min.js"`

pvnr0082t commented on ajax/libs/rita/package.json:
```diff
@@ -0,0 +1,33 @@
+  "npmName": "rita",
+  "npmFileMap": [
+    {
+      "basePath": "/dist",
```
There is no `dist` folder in npm package. I think we should use git auto-update here but I'm not sure. @PeterDaveHello Please give us some suggestions, thank you.

PeterDaveHello replied:
I didn't see `dist` in rita@1.1.391.

pvnr0082t commented:
@CyrusSUEN Please change the permission of files to inexecutable, thank you.

CyrusSUEN commented:
Thank you so much @pvnr0082t and @PeterDaveHello !
I've updated and pushed the changes. 

Please note that I've also changed the `url` attribute of `repository`  from https://github.com/dhowe/RiTa.git which is the Java version to https://github.com/dhowe/RiTaJS.git the JavaScript version.

PeterDaveHello commented:
@CyrusSUEN Thanks, we'll review your changes later once you passed the test.

PeterDaveHello commented:
@cdnjs/intern2 please review it.

CyrusSUEN commented:
Since only `rita.js` file is present in [npm package](https://www.npmjs.com/package/rita) while 4 dist files available in the [RiTaJS Github repo](https://github.com/dhowe/RiTaJS.git). 

Do you think I should configure it to git auto update?
If so, am I supposed to replace `npmFileMap` attribute in package.json with the following? Thanks so much again!
```json
  "autoupdate": {
    "source": "git",
    "target": "https://github.com/dhowe/ritajs",
    "basePath": "dist",
    "files": [
      "**/*"
    ]
  }
```

nolsherry commented:
@CyrusSUEN Could you help to add an [issue](https://github.com/dhowe/RiTaJS/issues) on github to report the problem about the difference between the files on npm and the files on github?

pvnr0082t commented on ajax/libs/rita/package.json:
```diff
@@ -0,0 +1,29 @@
+      ]
+    }
+  ],
+  "filename": "/rita.min.js"
```
Please remove the slash here.

PeterDaveHello replied:
@pvnr0082t you should at least modify our tests and the already exist libs to reflect this requirement.

CyrusSUEN replied:
> remove the slash here

Thanks! Slash removed.

> Please remove the slash here.

Thanks! Slash removed.

CyrusSUEN commented:
> report the problem about the difference between the files on npm and the files on github

I've notified the author. Not sure if that was an intended decision to put just one file on the npm package as I haven't heard back from him.
If that was the case, I will be sure to update the `package.json` to do git auto update instead.

CyrusSUEN commented:
The author wants to keep the download size of the npm package as small as possible. Changed to git auto update. Please review

PeterDaveHello commented on ajax/libs/rita/package.json:
```diff
@@ -0,0 +1,31 @@
+      "rita-full.js",
+      "rita-full.min.js",
+      "rita.js",
+      "rita.min.js"
```
Let's use single `"**/*"` to grab all the files here.

CyrusSUEN replied:
> single "**/*"

Thank you! Noted and updated.
